modules/docs/arrow-docs/docs/docs/datatypes/statet/README.md

```diff
@@ -136,7 +136,7 @@ import arrow.instances.either.monadError.*
 
 fun stackOperationsS2() = 
  ForStateT<EitherPartialOf<StackError>, Stack, StackError>(Either.monadError<StackError>()) extensions {
-  binding {
+  binding<StateTPartialOf<EitherPartialOf<StackError>, Stack>, String> {
```

Comment:
@pakoito seems I this message didn't go through or I just forgot 🤔 

Anyways, the inference trips here. Same for `StateApi.monad<Stack>().binding { }`. Also IntelliJ doesn't complain and the user ends up with a `Kotlinc Back-end error..". Thoughts?

Reply:
>Also IntelliJ doesn't complain and the user ends up with a `Kotlinc Back-end error..". Thoughts?

Async code and specially coroutines don't work on Ank. Hardcode the result instead.

Reply:
@pakoito

Reply:
If this is blocking you, hardcode the result.

Reply:
@pakoito I think I am not explaining it properly 😅

In Arrow 0.7.3
* `binding { ... }` compiles and the generic types `<F, B>` are correctly inferred.
* `State<S, A>` is an alias for `StateT<ForId, S, A>`

In Arrow 0.8.0
* `binding { ... }` **does not compile**, the compiler trips over `<F, B>` with a vague `kotlinc Back-end error..` message. This is fixed by explicitly specifying the generic types but IDEA marks them as redundant. (gray & underlined).
* `State<S, A>` is an alias for `IndexedStateT<ForId, S, S, A>`
* `IndexedState<ForId, SA, SB, A>` is an alias for `IndexedStateT<ForId, SA, SB, A>`
* `StateT<F, S, A>` is an alias for `IndexedStateT<F, S, S, A>`

In other words, this results in a very tedious migration for `State` from `0.7.3` to `0.8.0` since it results in a `kotlinc Back-end error..` and IDEA claims the types do not need to be specified for `binding`.